rust/arrow/src/compute/kernels/arithmetic.rs

```diff
@@ -35,20 +35,17 @@ use crate::datatypes;
 use crate::datatypes::ArrowNumericType;
 use crate::error::{ArrowError, Result};
 use crate::{array::*, util::bit_util};
+use num::traits::Pow;
 #[cfg(simd)]
 use std::borrow::BorrowMut;
 #[cfg(simd)]
 use std::slice::{ChunksExact, ChunksExactMut};
 
 /// Helper function to perform math lambda function on values from single array of signed numeric
 /// type. If value is null then the output value is also null, so `-null` is `null`.
-pub fn signed_unary_math_op<T, F>(
-    array: &PrimitiveArray<T>,
-    op: F,
-) -> Result<PrimitiveArray<T>>
+pub fn unary_math_op<T, F>(array: &PrimitiveArray<T>, op: F) -> Result<PrimitiveArray<T>>
```

Comment:
This simplification makes this identical to `crate::array::compute::kernels::arity::unary`, which was recently added.

Reply:
Right.. Then there is some duplication. What do you propose? Use the `crate::array::compute::kernels::arity::unary` for the logic in this model as wel? To me that seems more DRY.

Reply:
@nevi-me I removed the helper function and used the `arity::unary::kernel`